### ISSUE TITLE
[@snowpack/app-template-vue] - Final prod build throwing error 'Uncaught TypeError: (void 0) is not a function'

### ISSUE THREAD

**Issue by skylarrikin** [pikapkg/create-snowpack-app#140]
Hi all,

I am still very new to snowpack, so please bear with me if my issue seems trivial.

My setup:
Ubuntu: 20.04
npm: 6.14.5
node: v13.15.0

I created a Vue project using the following command:
`npx create-snowpack-app vue --template @snowpack/app-template-vue`

My `snowpack.config.json `looks like this:
```
{
  "extends": "@snowpack/app-scripts-vue",
  "scripts": {},
  "plugins": ["@snowpack/plugin-webpack"]
}
```

My `package.json` looks like this:
```
{
  "scripts": {
    "start": "snowpack dev",
    "build": "snowpack build",
    "test": "echo \"Vue 3.0 test support is still in progress.\" && exit 1"
  },
  "dependencies": {
    "vue": "^3.0.0-0"
  },
  "devDependencies": {
    "@babel/core": "^7.10.2",
    "@babel/preset-env": "^7.10.2",
    "@snowpack/app-scripts-vue": "^1.1.4",
    "@snowpack/plugin-webpack": "^1.2.0",
    "snowpack": "^2.5.0"
  }
}
```

The project works as expected when I do '`npm run start`', however when I do '`npm run build`', everything compiles as expected but when I try to run the bundled 'build' folder (`cd build && python3 -m http.server`), I get the following error:
`Uncaught TypeError: (void 0) is not a function`
![image](https://user-images.githubusercontent.com/66931958/84616113-55ac9c80-af0e-11ea-86b2-32df43ca7e0b.png)

Any clue as to what could be going wrong?

Thank you

**Comment by FredKSchott** [comment]
Does either of these fix it for you?
-  `npm run build -- --no-bundle` Skips the bundler, if the bug is in the bundler
-  `npm run build -- --no-treeshake` Skips treeshaking, if the bug is in our treeshaking algo

**Comment by skylarrikin** [comment]
Doing `npm run build -- --no-bundle `gives me a new error:
_localhost/:1 Uncaught SyntaxError: The requested module '/web_modules/vue.js' does not provide an export named 'openBlock'_

However, doing `npm run build -- --no-treeshake` works like a charm!

I believe I should keep this issue open until the bug in the treeshaking algorithm has been fixed, is that right?

Thanks Fred.

**Comment by FredKSchott** [comment]
Yes that's correct! This is a current issue in our tree-shake algo, hoping to fix it soon!

**Comment by djmisterjon** [comment]
same here 
![image](https://user-images.githubusercontent.com/24865815/84801790-9f0ff100-afcd-11ea-8620-be61163ebdcd.png)
`npm run build` crash !
![image](https://user-images.githubusercontent.com/24865815/84802195-3bd28e80-afce-11ea-813a-ebb5443c45bb.png)

**Comment by djmisterjon** [comment]
`npm run build -- --no-bundle` work
but no `npm run build -- --no-treeshake`

**Comment by FredKSchott** [comment]
You can also add this line to your snowpack.config:

```
{
  "install": ["vue"]
}
```

This tells Snowpack to install the full package, and will whitelist it for tree-shaking. With that, you should be able to run `npm run build` without flags.

**Comment by skylarrikin** [comment]
Thanks Fred.

One more issue that I noticed is that if I make changes to 'App.vue' such that it throws a script error, the entire hot loading mechanism ceases to work. Try changing the script code to as follows:
```
<script>
export default {
  data: {
    message: 'Hello Vue!'
  }
};
</script>
```
The above would generate an error.

Now, let us revert the code to original:
```
<script>
export default {
  data() {
    return {
      message: "Learn Vue"
    };
  }
};
</script>
```

Notice: Hot reloading stops working. You need to manually do a refresh (F5).
Is this expected behaviour?

**Comment by djmisterjon** [comment]
well the test was for react ! 
Trying bundles Parcel but not work ! 
I just wanted to test, I will stay on my current config.
Until everything is stable :)

I wanted to update this configuration to test the speed.
https://github.com/djmisterjon/DDBEV/tree/master/config
https://github.com/djmisterjon/DDBEV/blob/master/package.json
It a mix with Parcel,eslint,nwjs,electron,react,prettier.

**Comment by FredKSchott** [comment]
Re: HMR - it would make sense that HMR stops working, but you should still get livereload AKA an automatic full page reload whenever you make a change. That sounds like a bug in the HMR client, if anyone can take a look.
